x-pack/plugins/lens/public/shared_components/color_palette/color_ranges.tsx

```diff
@@ -0,0 +1,419 @@
+import type { FocusEvent } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFieldNumber,
+  EuiColorPicker,
+  EuiButtonIcon,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiColorPickerSwatch,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiTextColor,
+} from '@elastic/eui';
+import { ValueMaxIcon } from '../../assets/value_max';
+import { ValueMinIcon } from '../../assets/value_min';
+import { RelatedIcon } from '../../assets/related';
+import { DistributeEquallyIcon } from '../../assets/distribute_equally';
+import { getDataMinMax, getStepValue, isValidColor, roundValue } from '../coloring/utils';
+import type { CustomPaletteParamsConfig, ColorStop } from '../../../common';
+import { useDebouncedValue } from '../index';
+
+const DEFAULT_COLOR = '#6092C0';
+
+interface ColorRanges {
+  color: string;
+  start: number;
+  end: number;
+}
+
+interface ColorRangesProps {
+  colorRanges: ColorRanges[];
+  paletteConfiguration: CustomPaletteParamsConfig | undefined;
+  onChange: (colorStops: ColorStop[], upperMax: number) => void;
+  dataBounds: { min: number; max: number };
+}
+
+function areStopsValid(colorStops: Array<{ color: string; stop: number }>) {
+  return colorStops.every(({ color, stop }) => !Number.isNaN(stop) && isValidColor(color));
+}
+
+function reversePalette(colorRanges: ColorRanges[]) {
+  return colorRanges
+    .map(({ color }, i) => ({
+      color,
+      start: colorRanges[colorRanges.length - i - 1].start,
+      end: colorRanges[colorRanges.length - i - 1].end,
+    }))
+    .reverse();
+}
+
+export function ColorRanges(props: ColorRangesProps) {
+  const { colorRanges, onChange, dataBounds, paletteConfiguration } = props;
+  const [isValid, setValid] = useState(true);
+  const [isDisabledStart, setDisableStart] = useState(false);
+  const [isDisabledEnd, setDisableEnd] = useState(false);
+  const onChangeWithValidation = (newColorRanges: ColorRanges[]) => {
+    const colorStops = newColorRanges.map((colorRange) => {
+      return {
+        color: colorRange.color,
+        stop: colorRange.start && Number(colorRange.start),
+      };
+    });
+    const upperMax = Number(newColorRanges[newColorRanges.length - 1].end);
+    if (areStopsValid(colorStops)) {
+      onChange(colorStops, upperMax);
+    }
+  };
+  const { inputValue: localColorRanges, handleInputChange: setColorRanges } = useDebouncedValue({
+    onChange: onChangeWithValidation,
+    value: colorRanges,
+  });
+  const rangeType = paletteConfiguration?.rangeType || 'percent';
+  const shouldDisableAdd = Boolean(
+    paletteConfiguration?.maxSteps && localColorRanges.length >= paletteConfiguration?.maxSteps
+  );
+
+  const getColorRangeElem = (colorRange: ColorRanges, index: number, isLast = false) => {
+    const value = isLast ? colorRange.end : colorRange.start;
+    const showDelete = index !== 0 && !isLast;
+    const showEdit = !showDelete && isLast ? isDisabledEnd : isDisabledStart;
+    const dataTestPrefix = 'lnsPalettePanel';
```

Comment:
Can we make this a `prop` for the component as it was in the `color_stops` component previously?

Reply:
I don't understand why in was used as props previously. We just hardcode this on another level...

Reply:
In general it enables reuse of the component, being a shared component. the `lnsPalettePanel` can be a default value.

Reply:
done